lib/src/firestore.dart

```diff
@@ -348,6 +367,41 @@ class _FirestoreData {
     setProperty(nativeInstance, key, data);
   }
 
+  void setFieldValue(String key, FieldValue value) {
```

Comment:
Wondering if we can replace this method with two dedicated methods for each special value? E.g. `setServerTimestamp(String key)` and `deleteField(String key)`?

This would also allow to remove `fieldValueDeleted`, `fieldValueServerTimestamp` and the new `FieldValue` enum.

Reply:
one issue is that the user can read a DocumentData/UpdateData object before sending to firestore (or call toMap/fromMap). If I create a document data and call setServerTimestamp and right call toMap, what would be the value then? And I think calling fromMap(toMap) and toMap(fromMap) should lead to the same object content. Not sure if I'm clear here though...

Reply:
Also if we don't have the FieldValue enum (but yes it could be a class), how could this be used if I want to create a map and call fromMap?